### PR TITLE
2372, 23508 - redirect to direct pay, API and UI updates

### PR DESCRIPTION
**Pull request by hfekete**
*Issue #:2372* /bcgov/entity#2372
*Issue #:23508* /bcgov/entity#23508

*Description of changes:*
API updates:
- removed check for user being mandatory premium
- updated return from handler and from endpoint to include data about payment and invoice id (needed for cc)

UI updates:
- added ability to understand additional API return values
- redirecting UI to direct pay when payment is required

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: search-api/src/search_api/models/document_access_request.py [hunk 1]
```diff
@@ -126,7 +126,8 @@ def json(self):
             'submissionDate': self.submission_date.isoformat(),
             'expiryDate': self.expiry_date.isoformat() if self.expiry_date else None,
             'outputFileKey': self._output_file_key,
-            'submitter': self.submitter.display_name if self.submitter else None
+            'submitter': self.submitter.display_name if self.submitter else None,
+            'paymentToken': self.payment_token
```
Comment (kialj876) [inline search-api/src/search_api/models/document_access_request.py:130]: can you just add a comment here that this is the 'invoice id' from the pay db

File: search-api/tests/unit/request_handlers/test_document_access_request_handler.py [hunk 2]
```diff
@@ -45,7 +46,10 @@ def test_save_request(client, session, jwt, mocker):
     assert document_access_request.submitter.firstname == user.firstname
 
 
-def test_create_invoice(client, session, jwt, mocker):
+@pytest.mark.parametrize('test_name,mock_response,is_payment_completion_date_expected', [
+    ('test_pad_invoice', {'id': 123, 'paymentMethod': 'PAD'}, True),
+    ('test_pad_invoice', {'id': 123, 'paymentMethod': 'DIRECT_PAY'}, False)])
+def test_create_invoice(client, session, jwt, mocker, test_name, mock_response, is_payment_completion_date_expected):
```
Comment (kialj876) [inline search-api/tests/unit/request_handlers/test_document_access_request_handler.py:52]: nice

**Review by kialj876 — APPROVED** [review]
Nice work, this looks good 👍

**Review by seeker25 — APPROVED** [review]

File: search-api/src/search_api/request_handlers/document_access_request_handler.py [hunk 3]
```diff
@@ -70,18 +70,22 @@ def create_invoice(document_access_request: DocumentAccessRequest, user_jwt: Jwt
                                           header, business_json)
 
         if payment_response.status_code in (HTTPStatus.OK, HTTPStatus.CREATED):
-            payment_completion_date = datetime.utcnow()
+            is_pad = payment_response.json().get('paymentMethod') == 'PAD'
+            today_utc = datetime.utcnow()
```
Comment (seeker25) [inline search-api/src/search_api/request_handlers/document_access_request_handler.py:74]: `datetime.now(tz=timezone.utc)`
datetime.utcnow() = depreciated

File: search-api/src/search_api/services/validator.py [hunk 4]
```diff
@@ -37,10 +37,6 @@ def validate_document_access_request(document_access_request_json: dict, account
             if not account_org:
                 validation_errors.append({'error': 'Invalid Account'})
 
-            if account_org.get('orgType') != 'PREMIUM':
```
Comment (seeker25) [inline search-api/src/search_api/services/validator.py:40]: Woohoo!

**Review by seeker25 — APPROVED** [review]
LGTM

File: search-api/src/search_api/request_handlers/document_access_request_handler.py (hunk 3 above)
Comment (hfekete) [reply search-api/src/search_api/request_handlers/document_access_request_handler.py]: Yeah, pycharm shows the same !
I wasn't sure about updating it in just one place, but you are right. When changing the code, update it in place I changed if it does not break everything else :D

**Pull request merged**